lib/Conversation/CommentForm/CommentFormInput.js

```diff
@@ -0,0 +1,103 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { MentionsInput, Mention } from 'react-mentions';
+import Avatar from '../../Avatar';
+import AvatarInformation from '../../Avatar/AvatarInformation';
+
+class CommentFormInput extends Component {
+  static propTypes = {
+    inputValue: PropTypes.string,
+    handleOnChange: PropTypes.func.isRequired,
+    placeholder: PropTypes.string,
+    handleOnFocus: PropTypes.func.isRequired,
+    focusOnMount: PropTypes.bool,
+    onRowCountChange: PropTypes.func.isRequired,
+    users: PropTypes.arrayOf(PropTypes.shape()).isRequired
+  };
+
+  static defaultProps = {
+    inputValue: '',
+    placeholder: '',
+    focusOnMount: false
+  };
+
+  state = {
+    inputHeight: '0px'
+  };
+
+  componentDidMount() {
+    this.setState({ inputHeight: this.getInputHeight() }); // eslint-disable-line react/no-did-mount-set-state
+  }
+
+  getInputHeight = () => {
```

Comment:
Are we using any of this input height logic? I can't see state being used in the rending or in the prop function passed. It may be left overs from the `ExpandingTextArea` component?

Reply:
I think we may use `this.props.onRowCountChange` but I'm not sure.

Reply:
Rather than copying `ExpandingTextArea` logic we could use the render prop pattern?

For example, here is some logic for using the new `CommentFormInput` which is wrapped in a `ExpandingTextArea` component, which accepts a function so it doesn't render a text area but instead shares methods/state.

```
<ExpandingTextArea onRowCountChange={this.props.onRowCountChange}>
  (calculateRowCount, height) =>
    <CommentFormInput onChange={calculateRowCount} />
</ExpandingTextArea>
```

Obviously there is more detail than what I have described there but this would remove the need to copy logic and would keep the source of truth for expanding text areas in 1 location 😄

Reply:
We use onRowCountChange in the app for some positioning logic (that way the conversation doesn't overlap another conversation thats positioned directly below if the textarea expands).

The logic is quite different from expanding text area too, so its not really copied logic (although it achieves a similar result)